File: Problems/2125. Number of Laser Beams in a Bank/2125. Number of Laser Beams in a Bank.py
from typing import List
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        res = 0
        prev_count = 0
        cur_count = 0
        
        for row in bank:
            cur_count = row.count('1')
            if cur_count > 0:
                res += prev_count * cur_count
                prev_count = cur_count
        
        return res

File: Problems/2125. Number of Laser Beams in a Bank/2125. Number of Laser Beams in a Bank.java
class Solution {
    private int countOne(String s) {
        int res = 0; 
        for (char ch: s.toCharArray()) {
            res += ch-'0';
        }
        return res;
    }
    
    public int numberOfBeams(String[] bank) {
        int res = 0;
        int cur_count, prev_count = 0;
        
        for (String row : bank) {
            cur_count = countOne(row);
            if (cur_count > 0) {
                res += cur_count * prev_count;
                prev_count = cur_count;
            }
        }

        return res;
    }
}